connectors/sink-feishu/internal/bot.go

```diff
@@ -18,30 +18,216 @@ import (
 	"crypto/hmac"
 	"crypto/sha256"
 	"encoding/base64"
+	"encoding/json"
+	"errors"
 	"fmt"
-	"github.com/linkall-labs/cdk-go/log"
-	"github.com/tidwall/gjson"
 	"time"
 
 	v2 "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"github.com/go-resty/resty/v2"
+	"github.com/linkall-labs/cdk-go/log"
+	"github.com/tidwall/gjson"
 )
 
-func (f *feishuSink) sendTextToFeishuBot(e *v2.Event) error {
-	t := time.Now()
+type messageType string
+
+const (
+	xChatGroupID = "xvfeishuchatgroup"
+	xMessageType = "xvfeishumsgtype"
+
+	textMessage        = messageType("text")
+	postMessage        = messageType("post")
+	shareChatMessage   = messageType("share_chat")
+	imageMessage       = messageType("image")
+	interactiveMessage = messageType("interactive")
+)
+
+var (
+	errChatGroup = errors.New("feishu: xvfeishuchatgroup is missing or incorrect, please check your Feishu " +
+		"Sink config or subscription")
+	errMessageType = errors.New("feishu: xvfeishumsgtype is missing or invalid, only" +
+		" [text, post, share_chat, image, interactive] are supported")
+	errInvalidPostMessage = errors.New("feishu: invalid post message, please make sure it's the json format")
+)
+
+type bot struct {
+	cfg        BotConfig
+	cm         map[string]WebHook
+	httpClient *resty.Client
+}
+
+func (b *bot) init(cfg BotConfig) error {
+	b.cfg = cfg
+	b.cm = make(map[string]WebHook, len(cfg.Webhooks))
+	for _, wh := range cfg.Webhooks {
+		_, exist := b.cm[wh.ChatGroup]
+		if exist {
+			return fmt.Errorf("the chat_group has conflicted with name: %s", wh.ChatGroup)
+		}
+		b.cm[wh.ChatGroup] = wh
+	}
+	return nil
+}
+
+type WebHook struct {
+	Address   string `json:"address" yaml:"address"`
+	Signature string `json:"signature" yaml:"signature"`
+	ChatGroup string `json:"chat_group" yaml:"chat_group"`
+}
+
+func (wh WebHook) Validate() error {
+	if wh.Address == "" {
```

Comment:
It is no problem, but you also can add the tag `validate:"required"`

Reply:
It doesn't work in my local

Reply:
ok, I will test it in the cdk too.